tests/common/test_cgroupapi.py

```diff
@@ -505,11 +510,21 @@ def mock_popen(*args, **kwargs):
                         # No cgroups should have been created
                         self.assertEquals(extension_cgroups, [])
 
-    @patch("azurelinuxagent.common.cgroupapi.add_event")
-    def test_start_extension_command_should_use_fallback_option_if_systemd_times_out(self, *args):
+    @patch('time.sleep', side_effect=lambda _: mock_sleep)
+    def test_start_extension_command_should_use_fallback_option_if_systemd_times_out(self, _):
+        from azurelinuxagent.common.utils.extensionprocessutil import wait_for_process_completion_or_timeout
         # Mock systemd timeout and make sure the failure is only attributed to the extension if the command fails
         # using the fallback option
         original_popen = subprocess.Popen
+        mock_counter = Mock()
+        original_wait_for_completion = wait_for_process_completion_or_timeout
+
+        def mock_wait_for_process_completion(mock_counter, *args):
```

Comment:
Why was this mock needed instead of just patching the return value of `wait_for_process_completion_or_timeout`?

Reply:
Initially you were just mocking the `wait_for_process_completion_or_timeout` call in this file (cgroupapi.py) so the 2nd call was going to the `azurelinuxagent.common.utils.extensionprocessutil.wait_for_process_completion_or_timeout` and helping your test. 
Now we needed to first return True for timeout in the first run and get the output of the command in the 2nd run. This is why I had to mock this function like this to make sure we follow both the cases.

Reply:
Modified this test case, now inducing a timeout instead of mocking the return